From 24421ba92c69f1abaadaff01270621917616a2b8 Mon Sep 17 00:00:00 2001
From: Changqing Li <changqing.li@windriver.com>
Date: Thu, 17 Apr 2025 10:38:50 +0800
Subject: [PATCH] Fix build failures due to missing location.hh

ast/location.h does `#include "location.hh"` and location.hh is
generated by the parser so any CMake target whose source includes
ast/location.h needs to have a dependency on the `parser` target,
otherwise the compilation may fail due to incorrect ordering of build
targets. This also applies to targets which include ast/location.h
transitively via other headers.

To avoid such errors, do 2 things:
- drop includes of ast/location.h where unused
- for CMake targets including ast/location.h, add an explicit dependency
  on the `parser` target

Signed-off-by: Viktor Malik <viktor.malik@gmail.com>

Upstream-Status: Backport [https://github.com/bpftrace/bpftrace/commit/743cb9c88c25fb9737d714f4d4ac853f05bb6481]

Signed-off-by: Changqing Li <changqing.li@windriver.com>
---
 src/CMakeLists.txt     | 1 +
 src/ast/CMakeLists.txt | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e04ee429..13a51040 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -21,6 +21,7 @@ add_library(compiler_core STATIC
   struct.cpp
   types.cpp
 )
+add_dependencies(compiler_core parser)
 
 add_library(runtime STATIC
   attached_probe.cpp
diff --git a/src/ast/CMakeLists.txt b/src/ast/CMakeLists.txt
index 92231f85..7230cf71 100644
--- a/src/ast/CMakeLists.txt
+++ b/src/ast/CMakeLists.txt
@@ -22,6 +22,7 @@ add_library(ast STATIC
   passes/return_path_analyser.cpp
 )
 
+add_dependencies(ast parser)
 target_compile_definitions(ast PRIVATE ${BPFTRACE_FLAGS})
 target_link_libraries(ast PUBLIC ast_defs arch compiler_core parser)
 
-- 
2.34.1

